src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,72 @@
+package duke;
+
+import commands.*;
+import exceptions.DukeException;
+import tasks.*;
+
+public class Parser {
+
+    public static CommandType getCommandType(String input) {
+        for (CommandType c : CommandType.values()) {
+            if (c.equals(input)) {
+                return c;
+            }
+        }
+        return CommandType.UNKNOWN;
+    }
+
+    public static boolean isDescriptionEmpty(String[] arr) {
+        return arr.length == 1 || arr[1].trim().isEmpty();
+    }
+
+    public static Command parse(String input) throws DukeException {
+        String[] inputArr = input.split(" ", 2);
+        CommandType commandType = getCommandType(inputArr[0]);
+        int index = 0;
+        String taskInput = "";
+
+        switch (commandType) {
+            case LIST:
+                return new ListCommand();
+            case MARK:
+            case UNMARK:
+                index = Integer.parseInt(inputArr[1]) - 1;
+                return new MarkCommand(commandType, index);
+            case DELETE:
+                index = Integer.parseInt(inputArr[1]) - 1;
+                return new DeleteCommand(index);
+            case TODO:
+                if (isDescriptionEmpty(inputArr)) {
+                    throw new DukeException("Hey now.. The description of a todo cannot be empty. >:(");
+                }
+                taskInput = inputArr[1];
+                Todo todo = new Todo(taskInput);
+                return new AddCommand(CommandType.TODO, todo);
+            case DEADLINE:
+                if (isDescriptionEmpty(inputArr)) {
+                    throw new DukeException("Hey now.. The description of a deadline cannot be empty. >:(");
+                }
+                taskInput = inputArr[1];
+                String[] dArr = taskInput.split("/by", 2);
```

Comment:
The naming of this array might not be immediately evident on what it is supposed to be, because of the short name. Perhaps we can use a more precise name?